tla-import/src/main/scala/at/forsyte/apalache/tla/imp/ExprOrOpArgNodeTranslator.scala

```diff
@@ -103,6 +115,29 @@ class ExprOrOpArgNodeTranslator(sourceStore: SourceStore,
     LetInEx( body, letInDeclarations : _* )
   }
 
+  // translate an operator definition that is used as an expression, that is, LAMBDA
+  private def translateLambdaOrOperatorAsArgument(opArgNode: OpArgNode): TlaEx = {
+    // Instead of extending the IR with a new expression type, we simply introduce a local LET-IN definition.
+    // Although this is a well-defined expression in the IR, it does not correspond to a well-defined TLA+ expression.
+    // Hence, one has to take care of this, when printing the output to the user.
+    val name = opArgNode.getName.toString
+    opArgNode.getOp match {
+      // a lambda-definition is passed as an argument
+      case defNode: OpDefNode if name == "LAMBDA" =>
+        val decl = OpDefTranslator(sourceStore, context).translate(defNode)
+        // e.g., LET Foo(x) == e1 in Foo
+        LetInEx(NameEx(opArgNode.getName.toString), decl)
```

Comment:
Since you already have `name` in scope from line 123:

```suggestion
        LetInEx(name, decl)
```